Chrysalis-Papilio/Soen490-Papilio-Website
# Papilio 40 billing

Sure! Here's a paraphrased version of the PR description you provided, organized into three sections.

**GENERAL:**
This pull request adds new pages to handle billing and payments. The billing page has three sections: Subscription Information, Credit Card Information, and Payment History. The Payment page displays info of the selected package and allows the user to input their credit card information and choose to save it for future use.

**TASK DESCRIPTION:**
In the Subscription Information section of the billing page, the user can view their current active subscription, and cost, and due If there is no active subscription, it will display "No Active Subscription." The Credit Card Information section displays the card information for recurring payment processing. If there is no credit card it will display "No credit card information found". The Payment History section logs all payments placed. If there is no payment history , it will display "No payment history found" . Additionally, the select button for packages in the ad page has been edited to pass information about the selected package to the payment page. 

**MANUAL TEST:**
In order to test these changes, follow the steps below:

1. Run npm install (to install the credit card library).
2. Run npm start.
3. Navigate to the billing page and check the layout. currently there are no info retrieved. 
4. Navigate to the Ad center page , and select any package 
5. The selected package information should be passed through a URL to the payment page
6. On the Payment page verify the selected package information is displayed correctly
7. On the Payment page, input credit card information and check the "save card information" checkbox.

**Post-Backend Connection Plans for Implemented Features**

1. Troubleshoot and refactor passing the business ID to the backend once the payment is handled in order to retrieve the information in the correct billing account.
2. Improve the format and display of the retrieved information in the Subscription and Credit Card Information sections, such as recurring payments, display of the next cycle, display of any due balance, managing payment methods, deleting and adding credit cards, activating and deactivating recurring payments, deactivating subscriptions, and downgrading or upgrading subscriptions
